src/App.tsx

```diff
@@ -1,223 +1,179 @@
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
-  const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filterType, setFilterType] = useState<FilterType>(FilterType.All);
+  const [isLoading, setIsLoading] = useState(false);
+  const [title, setTitle] = useState('');
+  const [isSelectId, setIsSelectId] = useState<number[]>([]);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
-    if (newTodoField.current) {
-      newTodoField.current.focus();
+    const getTodoFromServer = async (userId: number) => {
+      try {
+        const todosFromServer = await getTodos(userId);
+
+        setTodos(todosFromServer);
+      } catch (error) {
+        setErrorMessage(`${error}`);
+      }
+    };
+
+    if (!user) {
+      return;
     }
+
+    getTodoFromServer(user.id);
   }, []);
 
+  const filteredTodos = useMemo(() => {
+    return todos.filter((todo) => {
+      switch (filterType) {
+        case FilterType.Active:
+          return !todo.completed;
+
+        case FilterType.Completed:
+          return todo.completed;
+
+        default:
+          return todo;
+      }
+    });
+  }, [todos, filterType]);
+
+  const isActiveTodos = useMemo(() => {
+    return todos.some(todo => !todo.completed);
+  }, [todos]);
+
+  const newTodo = useCallback(async (event: FormEvent) => {
+    event.preventDefault();
+
+    if (!user || !title) {
+      setErrorMessage(ErrorMessage.ErrorTitle);
+
+      return;
+    }
+
+    setIsLoading(true);
+
+    try {
+      const addedTodo = await addTodo(title, user.id);
+
+      setTodos([...todos, addedTodo]);
+    } catch {
+      setErrorMessage(ErrorMessage.NotAdd);
+    }
+
+    setIsLoading(false);
+    setTitle('');
+  }, [user, title]);
+
+  const removeTodo = useCallback(async (TodoId: number) => {
+    setIsSelectId([TodoId]);
+
+    try {
+      await deleteTodo(TodoId);
+      setTodos([...todos.filter(({ id }) => id !== TodoId)]);
+    } catch {
+      setErrorMessage(ErrorMessage.NotDelete);
+    }
+  }, [todos, errorMessage]);
+
+  const completedTodo = todos.filter(({ completed }) => !completed);
+
+  const deleteTodoCompleted = useCallback(() => {
+    setIsSelectId([...completedTodo].map(({ id }) => id));
+    Promise.all(completedTodo.map(({ id }) => removeTodo(id)))
+      .then(() => setTodos([...todos.filter(({ completed }) => !completed)]))
+      .catch(() => {
+        setErrorMessage(ErrorMessage.NotDelete);
+        setIsSelectId([]);
+      });
+  }, [todos, setIsSelectId, errorMessage]);
+
+  const changeStatus = useCallback(async (
+    todoId: number,
+    data: Partial<Todo>,
+  ) => {
+    setIsSelectId([todoId]);
+
+    try {
+      const changedTodo = await updateTodo(
+        todoId,
+        data,
+      );
+
+      setTodos([...todos.map(todo => (todo.id === todoId
```

Comment:
If the new state is computed using the previous state, you can pass a function to setTodos